File: EXERCICIOS/Assunto.c
#include <stdio.h>
#include <string.h>

int verificar(char s[]){
  int resp=0;
  int comecou=0;
  int assuntos=0;

for(int x=0; x<strlen(s); x++){

     if(s[x]=='('){
     assuntos++;
     comecou=1;
     }
     else if(comecou &&s[x]==')'){
        assuntos--;
     }

}

if(assuntos==0){
    resp=1;
}


return resp;
}

int main(){

int n;
char s[20];

scanf("%d", &n);
getchar();

for(int x=0; x<n; x++){
  scanf(" %[^\n]", s);

  if(verificar(s)){
    printf("S\n");
  }
  else{
    printf("N\n");
  }

}

    return 0;
}

File: EXERCICIOS/Assunto.java
import java.util.Scanner;

public class Assunto{
  
   public static boolean verificar(String s){
   boolean resp=false;
   boolean comecou=false;
   int assuntos=0;

   for(int x=0; x<s.length(); x++){
      if(s.charAt(x)=='('){
        assuntos++;
        comecou=true;
      }
      else if(comecou==true && s.charAt(x)==')'){
        assuntos--;
      }
   }

   if(assuntos==0){
    resp=true;
   }

  return resp;
   }
    public static void main(String args[]){

        Scanner sc = new Scanner(System.in);

        String entrada=sc.nextLine();

        if(verificar(entrada)){
            System.out.println("S");
        }
        else{
            System.out.println("N");
        }

        sc.close();
    }
}
